### PR TITLE
Handling units for Nexus hydraulics, gaslift and valve methods

### PR DESCRIPTION
**Pull request by obi-isebor**

### CODE REVIEW

**Review by cflynn3 — APPROVED** [review]
Looks good to me.

File: ResSimpy/Nexus/DataModels/NexusGasliftMethod.py [hunk 1]
```diff
@@ -28,16 +29,36 @@ class NexusGasliftMethod(DynamicProperty):
     file: NexusFile
     properties: dict[str, Union[str, int, float, Enum, list[str], pd.DataFrame,
                      dict[str, Union[float, pd.DataFrame]]]] = field(default_factory=get_empty_dict_union)
+    unit_system: UnitSystem
 
-    def __init__(self, file: NexusFile, input_number: int,
+    def __init__(self, file: NexusFile, input_number: int, model_unit_system: UnitSystem,
```
Comment (jonnyh404) [inline ResSimpy/Nexus/DataModels/NexusGasliftMethod.py:34]: We've called this argument "unit_system" in other places in the code. What's the reasoning behind calling this model_unit_system here?

File: ResSimpy/Nexus/DataModels/NexusValveMethod.py [hunk 2]
```diff
@@ -28,16 +29,31 @@ class NexusValveMethod(DynamicProperty):
     file: NexusFile
     properties: dict[str, Union[str, int, float, Enum, list[str], pd.DataFrame,
                      dict[str, Union[float, pd.DataFrame]]]] = field(default_factory=get_empty_dict_union)
+    unit_system: UnitSystem
 
-    def __init__(self, file: NexusFile, input_number: int,
+    def __init__(self, file: NexusFile, input_number: int, model_unit_system: UnitSystem,
```
Comment (jonnyh404) [inline ResSimpy/Nexus/DataModels/NexusValveMethod.py:34]: Same model_unit_system vs just unit_system?

File: ResSimpy/Nexus/DataModels/NexusValveMethod.py [hunk 3]
```diff
@@ -28,16 +29,31 @@ class NexusValveMethod(DynamicProperty):
...
         else:
             self.properties = {}
+        self.unit_system = model_unit_system
         super().__init__(input_number=input_number, file=file)
 
+    @staticmethod
+    def get_keyword_mapping() -> dict[str, tuple[str, type]]:
+        """Gets the mapping of nexus keywords to attribute definitions."""
+        keywords: dict[str, tuple[str, type]] = {
+            'VC': ('valve_coefficient', float),
```
Comment (jonnyh404) [inline ResSimpy/Nexus/DataModels/NexusValveMethod.py:48]: Is this really all the keywords in valves?

File: tests/Nexus/Properties/test_nexus_gaslift_class.py [hunk 4]
```diff
@@ -96,3 +98,54 @@ def test_nexus_gaslift_repr():
...
+WCUT 0.0 0.2 0.4
+QLIQ 1000 3500
+PRESSURE 2500 4500
+""" + properties['GL_TABLE'].to_string(na_rep='', index=False) + '\n\n\n'
+
+    # Act
+    result = gl_methods_obj.__repr__()
+
+    # Assert
+    assert result == expected_output
```
Comment (jonnyh404) [inline tests/Nexus/Properties/test_nexus_gaslift_class.py:151]: Good tests for these repr methods. We should get these sorts of repr for the NexusSimulator class and network objects.

**Review by jonnyh404 — APPROVED** [review]
Looks good to me, please see comments before merging.

**Pull request merged**